plugins/pipelines/get.js

```diff
@@ -20,9 +20,11 @@ module.exports = () => ({
 
         handler: async (request, h) => {
             const factory = request.server.app.pipelineFactory;
+            const { canAccessPipeline } = request.server.plugins.pipelines;
+            const { credentials } = request.auth;
 
-            return factory
-                .get(request.params.id)
+            return canAccessPipeline(credentials, request.params.id, 'pull', request.server.app)
```

Comment:
I think `canAccessPipeline` should take pipeline object as 2nd argument instead of pipeline id to prevent duplicate pipeline fetch from DB